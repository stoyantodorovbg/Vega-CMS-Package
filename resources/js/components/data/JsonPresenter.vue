<template>
    <div>
        <div v-for="(value, key) in json_data" :key="key">
            {{ key }}: {{ value }}
        </div>
    </div>
</template>

<script>
    export default {
        name: 'JsonPresenter',

        props: ['json_data'],

        // props: {
        //     json_data: {
        //         type: Object,
        //         default: function () {
        //             return {
        //                 classes: '',
        //                 status: 1,
        //                 text: 'data',
        //                 styles: [],
        //                 structure: {
        //                     classes: {
        //                         type: 'text',
        //                     },
        //                     status: {
        //                         type: 'text',
        //                     },
        //                     styles: {
        //                         nested: [],
        //                         type: 'json'
        //                     },
        //                     text: {
        //                         type: 'text',
        //                     },
        //                 },
        //             };
        //         }
        //     }
        // },
    }
</script>
