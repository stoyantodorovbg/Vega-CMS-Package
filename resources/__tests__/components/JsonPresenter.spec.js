import { mount } from '@vue/test-utils';
import JsonPresenter from "../../js/components/data/JsonPresenter";

describe('JsonPresenter Component', () => {
    test('should mount without crashing', () => {
      const wrapper = mount(JsonPresenter);
      expect(wrapper).toMatchSnapshot();
    });
});
