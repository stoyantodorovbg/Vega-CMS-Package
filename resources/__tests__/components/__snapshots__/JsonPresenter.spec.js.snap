// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonPresenter Component should mount without crashing 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
